repose-aggregator/components/filters/api-validator-filter/src/main/java/org/openrepose/filters/apivalidator/ApiValidatorHandler.java

```diff
@@ -153,9 +146,16 @@ public void doFilter(HttpServletRequest request, HttpServletResponse response, F
                 }
 
                 if (!isValid) {
-                    if (mbcsInvalidRequests != null) {
+                    if (Optional.ofNullable(metricsService).isPresent()) {
```

Comment:
So after a quick bit of research, it appears that Spring will actually blow up if a dependency is not satisfied. Although, that behavior can be controlled using the `required` parameter on the `Autowired` annotation.

Anyway, technically, core services as we have defined them are always available. Also technically, the metrics service is currently a core service. So right now it's probably safe to drop the null checks, but once we move to services as first class citizens, then all bets are off. I hope we can come up with a design where we don't have `Optional`s all over the place to handle optional services. I mean, that isn't the worst, but it's also not the best. The only general solution I can think of involves reflection and dynamic method call routing, but that's about as nasty as it sounds. Like, worse than the JAX-RS `RuntimeDelegate` nasty, because at least that has a static interface.